src/libs/firebase/firestore/featureFlags/useFeatureFlag.ts

```diff
@@ -6,18 +6,32 @@ import { getAppBuildVersion } from 'libs/packageJson'
 
 // firestore feature flag documentation :
 // https://www.notion.so/passcultureapp/Feature-Flag-e7b0da7946f64020b8403e3581b4ed42#fff5fb17737240c9996c432117acacd8
-export const useFeatureFlag = (featureFlag: RemoteStoreFeatureFlags): boolean | undefined => {
-  const [minimalBuildNumber, setMinimalBuildNumber] = useState<number | null>()
+export const useFeatureFlag = (
+  remoteStorefeatureFlag: RemoteStoreFeatureFlags
+): boolean | undefined => {
+  const [buildNumberConfig, setBuildNumberConfig] = useState<{
+    minimalBuildNumber?: number | null
+    maximalBuildNumber?: number | null
```

Comment:
les valeurs étant encapsulés, je ne vois pas l'intéret d'avoir 2 types pour exprimer l'absence de valeur

```suggestion
    minimalBuildNumber?: number
    maximalBuildNumber?: number
```

Reply:
Le `undefined` c'est pour dire qu'on n'a toujours pas la réponse de firestore, le `null` c'est quand aucune valeur n'est précisé sur firestore

Reply:
en cas d'égalité, on veut que ça soit activé, donc on pourrait carrement avoir

```suggestion
    minimalBuildNumber: number
    maximalBuildNumber: number
```

si pas de version, fallback sur la version courante

je ne sais pas si c'est une bonne idée

Reply:
on n'expose pas cette information hors du hook

on a le meme comportement entre "on n'a pas encore l'info" et "on n'a pas l'info"

est-ce qu'il y a une utilité à les séparer ?

Reply:
proposition (pas testé, écrit à l'arrache)

```ts
export const useFeatureFlag = (remoteStorefeatureFlag: RemoteStoreFeatureFlags): boolean => {
  const appBuildVersion = getAppBuildVersion()
  const [{ minimalBuildNumber, maximalBuildNumber }, setBuildNumberConfig] = useState<{
    minimalBuildNumber: number
    maximalBuildNumber: number
  }>({
    minimalBuildNumber: appBuildVersion,
    maximalBuildNumber: appBuildVersion,
  })

  useEffect(() => {
    async function updateFeatureFlag() {
      const featureFlag = await getFeatureFlag(remoteStorefeatureFlag)
      setBuildNumberConfig({
        minimalBuildNumber: featureFlag?.minimalBuildNumber || appBuildVersion,
        maximalBuildNumber: featureFlag?.maximalBuildNumber || appBuildVersion,
      })
    }

    updateFeatureFlag()
  }, [appBuildVersion, remoteStorefeatureFlag])

  return minimalBuildNumber <= appBuildVersion && appBuildVersion <= maximalBuildNumber
}
```

Reply:
j'ai oublié le cas où il faut lever une erreur

```ts
if (minimalBuildNumber > maximalBuildNumber) error()
```

Reply:
Avec cette implém, les cas suivants rentrent en conflit : 
- Si minimal = maximal, on veut désactiver le FF
- Si maximal ou minimal = version actuel, on veut activer le FF

Reply:
> Si minimal = maximal, on veut désactiver le FF

pour quelles raisons voudrait-on faire ça ?

si on veut activer un FF seulement sur une version

---

> Si maximal ou minimal = version actuel, on veut activer le FF

avec le code que j'ai proposé, j'ai l'impression que c'est le cas

Reply:
et si c'est pas le cas, ça me semble plus facile d'expliciter avec une condition en amont qui montre ce edge case

Reply:
> pour quelles raisons voudrait-on faire ça ?
C'était marqué dans la TODO

> avec le code que j'ai proposé, j'ai l'impression que c'est le cas
Oui et non, par exemple si sur firestore on met minimal = version actuelle, et qu'on ne précise pas de version maximale, le FF sera désactivé (parce que la valeur par défaut de maximal sera version actuelle, du coup on a minimal = maximal, et dans ce cas on désactive le FF d'après la TODO)